akka-stream-tests/src/test/scala/akka/stream/scaladsl/FlowFoldAsyncSpec.scala

```diff
@@ -0,0 +1,219 @@
+/**
+ * Copyright (C) 2014-2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.stream.scaladsl
+
+import scala.util.control.NoStackTrace
+
+import scala.concurrent.{ Await, Future }
+import scala.concurrent.duration._
+
+import akka.NotUsed
+import akka.testkit.{ AkkaSpec, TestLatch }
+import akka.stream.ActorMaterializer
+import akka.stream.ActorAttributes.supervisionStrategy
+import akka.stream.Supervision.resumingDecider
+import akka.stream.testkit._, Utils._
+
+class FlowFoldAsyncSpec extends StreamSpec {
+  implicit val materializer = ActorMaterializer()
+  implicit def ec = materializer.executionContext
+
+  "A FoldAsync" must {
+    val input = 1 to 100
+    val expected = input.sum
+    val inputSource = Source(input).filter(_ ⇒ true).map(identity)
+    val foldSource = inputSource.foldAsync[Int](0) { (a, b) ⇒
+      Future(a + b)
+    }.filter(_ ⇒ true).map(identity)
+    val foldFlow = Flow[Int].filter(_ ⇒ true).map(identity).foldAsync(0) {
+      (a, b) ⇒ Future { Thread.sleep(100); a + b }
+    }.filter(_ ⇒ true).map(identity)
+    val foldSink = Sink.foldAsync[Int, Int](0) { (a, b) ⇒ Future(a + b) }
+
+    "work when using Source.foldAsync" in assertAllStagesStopped {
+      Await.result(foldSource runWith Sink.head, 3.seconds) should be(expected)
+    }
+
+    "work when using Sink.foldAsync" in assertAllStagesStopped {
+      Await.result(inputSource runWith foldSink, 3.seconds) should be(expected)
+    }
+
+    "work when using Flow.foldAsync" in assertAllStagesStopped {
+      Await.result(inputSource via foldFlow runWith Sink.head, 12.seconds) should be(expected)
```

Comment:
This test case takes > 10 seconds to complete, please make it as fast as the other Flow API tests.